meshpy/mesh_creation_functions/beam_basic_geometry.py

```diff
@@ -482,3 +483,171 @@ def create_beam_mesh_arc_at_node(
+                + "Height of helix can not be determined through turns! "
+                + "Either switch to height of helix or change radius!"
+            )
+
+        warnings.warn(
+            "Radius of helix is 0 or twist angle is 90 degrees! "
+            + "Simple line geometry is returned!"
+        )
+
+        return create_beam_mesh_line(
+            mesh,
+            beam_object,
+            material,
+            start_point=start_point,
+            end_point=start_point + height_helix * axis,
+            **kwargs,
+        )
+
+    # generate simple helix
+    if height_helix:
+        end_point = np.array([radius, height_helix / np.tan(twist_angle), height_helix])
+    elif turns:
+        end_point = np.array(
+            [
+                radius,
+                2 * np.pi * radius * turns,
+                2 * np.pi * radius * turns * np.tan(twist_angle),
+            ]
+        )
+
+    helix = create_beam_mesh_line(
+        mesh,
+        beam_object,
+        material,
+        start_point=[radius, 0, 0],
+        end_point=end_point,
+        **kwargs,
+    )
+
+    mesh.wrap_around_cylinder()
+
+    # transform simple helix to align with neccessary axis and starting point
+    axis_simple = np.array([0, 0, 1])
+    start_point_origin_vec_simple = np.array([1, 0, 0])
+
+    start_point_origin_vec /= np.linalg.norm(start_point_origin_vec)
+
+    # 1 - get rotation matrix to align axes
+    if np.isclose(np.dot(axis, axis_simple), 1):
+        # axis point in same direction
+        R_axis = np.eye(3)
+    elif np.isclose(np.dot(axis, axis_simple), -1):
+        # axis point in opposite direction
+        R_axis = np.asarray([[1, 0, 0], [0, -1, 0], [0, 0, -1]])
+    else:
+        v = np.cross(axis, axis_simple)
+        c = np.dot(axis, axis_simple)
+        v_x = np.asarray([[0, -v[2], v[1]], [v[2], 0, -v[0]], [-v[1], v[0], 0]])
+        R_axis = np.eye(3) + v_x + np.matmul(v_x, v_x) / (1 + c)
+
+    start_point_origin_vec_simple_rot = np.dot(R_axis, start_point_origin_vec_simple)
+    start_point_origin_vec_simple_rot /= np.linalg.norm(
+        start_point_origin_vec_simple_rot
+    )
+
+    # 2 - get rotation matrix to align start_point_origin_vec's
+    if np.isclose(np.dot(start_point_origin_vec, start_point_origin_vec_simple_rot), 1):
+        # vectors point in same direction
+        R_start_vec = np.eye(3)
+    elif np.isclose(
+        np.dot(start_point_origin_vec, start_point_origin_vec_simple_rot), -1
+    ):
+        # vectors point in opposite direction
+        R_start_vec = np.asarray([[-1, 0, 0], [0, 1, 0], [0, 0, -1]])
+    else:
+        v = np.cross(start_point_origin_vec, start_point_origin_vec_simple_rot)
+        c = np.dot(start_point_origin_vec, start_point_origin_vec_simple_rot)
+        v_x = np.asarray([[0, -v[2], v[1]], [v[2], 0, -v[0]], [-v[1], v[0], 0]])
+        R_start_vec = np.eye(3) + v_x + np.matmul(v_x, v_x) / (1 + c)
+
```

Comment:
Here, I need to determine the rotation between the simple helix and the wanted helical geometry. I used the approach described here: https://math.stackexchange.com/questions/180418/calculate-rotation-matrix-to-align-vector-a-to-vector-b-in-3d. There's definitely a shorter and more elegant solution than the long and elaborate way I did it using quaternions. I was not able to find something predefined within meshpy so if someone has a suggestion on how to improve this I would be more than happy to know.

Reply:
I think this can be simplified drastically:
```python
rotation = Rotation.from_basis(axis, start_point_origin_vec)*Rotation([0,1,0],np*0.5)*Rotation([0,0,1],np.pi*0.5)
mesh.rotate(rotation)
```
Be aware of the singularity if `norm(start_point_origin_vec) == 0`, related to the question above regarding the case `radius == 0`.

Reply:
Wow that's a really nice solution! See my comment down below.

Reply:
Whoops - wasn't aware that at Github the comments are connected and if i resolve the first one - this one is resolved as well - Sorry!

I've actually simplified your approach further to only use one rotation in the latest commit. Just one theoretical question: Here, we get the rotation necessary to rotate the helical coordinate system into the global "simple" coordinate system. Why do we not need to inverse the rotation/rotation matrix here? Initially I thought we need to reverse the rotation because we transform the simple geometry into the defined axis/start point.

Reply:
The way you initially create the helix, the helix axis will be in the z-direction and the starting point will lie on the x-axis. Thus we need to create a rotation that maps the z-axis onto the desired helix axis and the x-axis onto the `start_point_origin_vec`. No need to invert anything here.